src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -7981,16 +7988,28 @@ def astype(
             ErrorMessage.not_implemented(
                 f"Snowpark pandas astype API doesn't yet support errors == '{errors}'"
             )
-        col_dtypes_curr = {
-            k: v for k, v in self.dtypes.to_dict().items() if k in col_dtypes_map
-        }
+
+        col_dtypes_curr = {}
+        if include_index:
```

Comment:
@sfc-gh-vbudati since you haven't merged this pr yet, how about let's just do consistent on all implemention on index API, have a different query compiler API for the index like index_astype, turn this funciton into internal function to use

Reply:
done!